gpMgmt/test/behave/mgmt_utils/steps/mgmt_utils.py

```
@@ -2319,13 +2319,17 @@ def impl(context, table_name):
 def impl(context, num_of_segments):
     dbname = 'gptest'
     with dbconn.connect(dbconn.DbURL(dbname=dbname)) as conn:
-        query = """SELECT count(*) from gp_segment_configuration where -1 < content and status = 'u'"""
-        end_data_segments = dbconn.execSQLForSingleton(conn, query)
+        query = """SELECT dbid, content, role, preferred_role, mode, status, port, hostname, address, datadir from gp_segment_configuration;"""
+        seg_conf_result = dbconn.execSQL(conn, query).fetchall()
+        end_data_segments = 0
+        for row in seg_conf_result:
```

Can we have this logic placed in a WHERE clause for the query above like before?

---

When we need to raise the exception we want to dump the same and original information as the checks here, in such a case we should not query `gp_segment_configuration` twice, so we must obtain the entire unfiltered table above and check it manually here.

---

I should have been more clear. I meant that we could modify the query above as:
`query = """SELECT count(*), dbid, content, role, preferred_role, mode, status, port, hostname, address, datadir from gp_segment_configuration where -1 < content and status = 'u'"""`
Then we don't need to loop through the result and/or calculate the `end_data_segments` variable.

---

Well, the purpose of this PR is to dump full content of the `gp_segment_configuration` on exceptions, that includes not only the `-1 < content and status = 'u'` segments but also the other segments.  We want to prevent a second query to the catalog table to ensure that the dumped information is exactly the same as the information used in the checks.